### DISCUSSION TITLE
Connect 2 spine pairs

### DISCUSSION THREAD

**Discussion by PieterL75** [aristanetworks/avd#4433] [General]
We have a spine pairs per end of row.
Currently everything is configured manually. 

The spine pairs of row AB peer with the spine pairs in row CD.
There is one leaf that also connects to all four spines, these are our DCIs

Can I model this in avd?
I already have one row modeled out (2xspine and some leafs)

**Comment by ClausHolbechArista** [comment]
This should be doable by setting `uplink_switches`, `uplink_switch_interfaces` and `uplink_interfaces` correctly. On the DCI Leafs you should set a separate `uplink_ipv4_pool` to ensure no overlap in IPs. And same on each DC, they can use independent ID numbering and their own `uplink_ipv4_pool`

**Reply by PieterL75** [reply]
For now, I defined just one datacenter, with 4 spine nodes, and then the dc's in the l3leafs.
This way, I can connect the DCI's to all 4 spines by defining them all 4 in the uplink_switches. Drawback is that I had to increase the IP pool, because it uses a mod4 in stead of mod2 to create the p2p subnets.

Not sure how I can create the spine-spine uplinks now... 
I tried using the 'l3_edge' fabric setting, but that just generates the underlay config, and not the evpn connection...

any suggestion there ? or should I not peer on the overlay between the two spine core, as that is done on the DCI

**Reply by ClausHolbechArista** [reply]
Your topology is not clear to me. As you describe here, make sure to set `max_uplink_switches` to 4, to avoid overlapping IP addressing.

I think I would let the DCI nodes here be EVPN servers for the entire DC and then have the DCI nodes to peer with the DCI nodes in the other DC, maybe with the EVPN Gateway feature. So the spines would not be part of EVPN at all.

For underlay connectivity between the spines pairs you can use l3_edge or core_interfaces and remember to set `include_in_underlay_protocol`. This will *not* touch anything around overlay. Alternatively you can set `uplink_switches` on spine C+D to be spine A+B.
